### PR TITLE
Use format(result) instead of result

### PR DESCRIPTION
**Pull request by krlmlr**
when printing expectations.

### CODE REVIEW

File: R/expectation.R [hunk 1]
```diff
@@ -136,15 +136,9 @@ as.expectation.error <- function(x, ..., srcref = NULL) {
...
-    user_calls <- paste0(traceback, collapse = "\n")
-    msg <- paste0(msg, "\n", user_calls)
-  } else {
-    # Need to remove trailing newline from error message to be consistent
-    # with other messages
-    msg <- gsub("\n$", "", msg)
-  }
+  # Need to remove trailing newline from error message to be consistent
+  # with other messages
+  msg <- gsub("\n$", "", msg)
```
Comment (hadley) [inline R/expectation.R]: Should this also go in `format()`?

File: R/expectation.R [hunk 2]
```diff
@@ -178,10 +172,15 @@ print.expectation <- function(x, ...) cat(format(x), "\n")
 #' @export
 format.expectation <- function(x, ...) {
```
Comment (hadley) [inline R/expectation.R]: I wonder if we should a separate method for `success` (and I wonder if the class names should actually `expectation_success()`, `expectation_skipped()` etc)

Comment (krlmlr) [inline R/expectation.R]: format.success() will never be called if format.expectation() exists.

Comment (hadley) [inline R/expectation.R]: Then we have our classes around the wrong way

File: R/expectation.R (hunk 1 above)
Comment (krlmlr) [inline R/expectation.R]: Don't know -- this was code specific to error conditions.

**Pull request merged**